### PR TITLE
KREST-8294 Permit charset for v3 produce

### PR DESCRIPTION
**Pull request by AndrewJSchofield**
This permits a Content-Type header that includes UTF-8 or ISO-8859-1 as a charset header parameters such as `Content-Type=application/json; charset=UTF-8`.

### CODE REVIEW

File: kafka-rest/src/main/java/io/confluent/kafkarest/response/JsonStreamMessageBodyReader.java [hunk 1]
```diff
@@ -41,7 +42,9 @@ public JsonStreamMessageBodyReader(ObjectMapper objectMapper) {
   @Override
   public boolean isReadable(
       Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
-    return JsonStream.class.equals(type) && MediaType.APPLICATION_JSON_TYPE.equals(mediaType);
+    return JsonStream.class.equals(type)
+        && MediaType.APPLICATION_JSON_TYPE.getType().equalsIgnoreCase(mediaType.getType())
+        && MediaType.APPLICATION_JSON_TYPE.getSubtype().equalsIgnoreCase(mediaType.getSubtype());
```
Comment (trnguyencflt) [inline kafka-rest/src/main/java/io/confluent/kafkarest/response/JsonStreamMessageBodyReader.java:47]: Not sure if this is correct but I think the third condition should be optional if there is subtype specified, something like
`&& (mediaType.getSubtype() == null || MediaType.APPLICATION_JSON_TYPE.getSubtype().equalsIgnoreCase(mediaType.getSubtype()))`

File: kafka-rest/src/main/java/io/confluent/kafkarest/response/JsonStreamMessageBodyReader.java [hunk 2]
```diff
@@ -41,7 +42,9 @@ public JsonStreamMessageBodyReader(ObjectMapper objectMapper) {
   @Override
   public boolean isReadable(
       Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
-    return JsonStream.class.equals(type) && MediaType.APPLICATION_JSON_TYPE.equals(mediaType);
+    return JsonStream.class.equals(type)
```
Comment (trnguyencflt) [inline kafka-rest/src/main/java/io/confluent/kafkarest/response/JsonStreamMessageBodyReader.java:45]: For my curiosity, why we don't use .equals method of MediaType class?

File: kafka-rest/src/main/java/io/confluent/kafkarest/response/JsonStreamMessageBodyReader.java [hunk 3]
```diff
@@ -53,6 +56,18 @@ public JsonStream<?> readFrom(
       MultivaluedMap<String, String> httpHeaders,
       InputStream entityStream)
       throws WebApplicationException {
+
+    // We know that the media type is 'application/json' but there might be an optional charset too
+    if (mediaType.getParameters() != null) {
+      String charset = mediaType.getParameters().get(MediaType.CHARSET_PARAMETER);
```
Comment (trnguyencflt) [inline kafka-rest/src/main/java/io/confluent/kafkarest/response/JsonStreamMessageBodyReader.java:62]: should we use `withCharset` method instead

File: kafka-rest/src/main/java/io/confluent/kafkarest/response/JsonStreamMessageBodyReader.java [hunk 4]
```diff
@@ -53,6 +56,18 @@ public JsonStream<?> readFrom(
...
+      if (charset != null) {
+        if (!charset.equalsIgnoreCase("UTF-8") && !charset.equalsIgnoreCase("ISO-8859-1")) {
+          throw new WebApplicationException(
+              "Unsupported charset in Content-Type header.", Status.BAD_REQUEST);
```
Comment (trnguyencflt) [inline kafka-rest/src/main/java/io/confluent/kafkarest/response/JsonStreamMessageBodyReader.java:66]: Should we also provide the list of valid charsets in the exception?

Comment (AndrewJSchofield) [reply kafka-rest/src/main/java/io/confluent/kafkarest/response/JsonStreamMessageBodyReader.java:66]: That probably is a good point.

**Review by trnguyencflt — APPROVED** [review]
Approved with a very minor suggestion

**Pull request merged**